trin-core/src/portalnet/overlay_service.rs

```diff
@@ -3305,14 +3314,9 @@ mod tests {
 
         service.handle_find_content_query_event(query_event);
 
-        match callback_rx
+        let response = callback_rx
             .await
-            .expect("Expected result on callback channel receiver")
-        {
-            Some(result_content) => {
-                assert_eq!(result_content, content);
-            }
-            _ => panic!("Unexpected find content query result type"),
-        }
+            .expect("Expected result on callback channel receiver");
+        assert_eq!(response.0.unwrap(), content);
```

Comment:
I would prefer that we keep the original `match` format here. It's easier to understand what the test case asserts.